File: extensions/fixtures/wasmer-quote/quote.php
<?php
// Fixed Wasmer sandbox guest: quote(cart) -> adjustments (JSON stdin/stdout).
function quote(array $cart): array
{
    $currency = $cart['currency'];
    $adjustments = [];
    foreach ($cart['lines'] ?? [] as $line) {
        $quantity = (int) $line['quantity'];
        $unitMinor = (int) $line['unit_price']['amount_minor'];
        if ($quantity >= 2) {
            $adjustments[] = [
                'label' => 'volume-discount',
                'amount_minor' => intdiv(-($quantity * $unitMinor), 10),
                'currency' => $currency,
            ];
        }
    }
    return $adjustments;
}

$raw = stream_get_contents(STDIN);
$cart = json_decode($raw, true, 512, JSON_THROW_ON_ERROR);
echo json_encode(quote($cart), JSON_THROW_ON_ERROR);

File: extensions/fixtures/wasmer-quote/quote.py
# Fixed Wasmer sandbox guest: quote(cart) -> adjustments (JSON stdin/stdout).
import json
import sys


def quote(cart):
    currency = cart["currency"]
    adjustments = []
    for line in cart.get("lines", []):
        quantity = int(line["quantity"])
        unit_minor = int(line["unit_price"]["amount_minor"])
        if quantity >= 2:
            amount_minor = -(quantity * unit_minor) // 10
            adjustments.append(
                {
                    "label": "volume-discount",
                    "amount_minor": amount_minor,
                    "currency": currency,
                }
            )
    return adjustments


if __name__ == "__main__":
    cart = json.load(sys.stdin)
    json.dump(quote(cart), sys.stdout)
